flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/LazyFromSourcesSchedulingStrategy.java

```diff
@@ -102,11 +108,12 @@ public void onExecutionStateChange(ExecutionVertexID executionVertexId, Executio
 		final Set<SchedulingExecutionVertex> verticesToSchedule = schedulingTopology.getVertexOrThrow(executionVertexId)
 			.getProducedResultPartitions()
 			.stream()
+			.filter(partition -> partition.getPartitionType() == ResultPartitionType.BLOCKING)
```

Comment:
Is this a different bug?

Reply:
Yes. 

Here's some more details:
Technically this is a bug because it may schedule a task that is already scheduled in `onPartitionConsumable`.
However, it's not problematic at the moment because scheduling of a non-CREATED vertex will be filtered out with `IS_IN_CREATED_EXECUTION_STATE`.

Reply:
Isn't it safer to use `partition.getPartitionType().isBlocking()`?

Reply:
You are right!

Reply:
Done.